hls-plugin-api/src/Ide/Types.hs

```diff
@@ -383,168 +406,201 @@ class HasTracing (MessageParams m) => PluginMethod (k :: MessageKind) (m :: Meth
     -> Config
     -- ^ Generic config description, expected to contain 'PluginConfig' configuration
     -- for this plugin
-    -> Bool
+    -> PluginStatus
     -- ^ Is this plugin enabled and allowed to respond to the given request
     -- with the given parameters?
 
   default pluginEnabled :: (L.HasTextDocument (MessageParams m) doc, L.HasUri doc Uri)
-                              => SMethod m -> MessageParams m -> PluginDescriptor c -> Config -> Bool
-  pluginEnabled _ params desc conf = pluginResponsible uri desc && plcGlobalOn (configForPlugin conf desc)
+                              => SMethod m -> MessageParams m -> PluginDescriptor c -> Config -> PluginStatus
+  pluginEnabled _ params desc conf =
```

Comment:
I also think we should rename this method...

Reply:
Or alternatively, I think it would be reasonable to have `pluginEnabled` tell us the "global" enabled/disabled status (and hence doesn't look at the current request), and then have `pluginCanRespond` which checks if the plugin is disabled and also anything that depends on the current request.

Reply:
It seems like we don't need to provide a way to other parts of the code to determine whether a plugin is enabled or not here. That actually should just use plcGlobalOn. In addition, determining whether a feature is enabled should use the specific field for that. Hence I have changed the two instances in the plugins that relied on it, and have also renamed it to the more correct "handlesRequest"

Reply:
So it seems like the specific case we have is for diagnostics produced by a plugin, where we need to check that the plugin is enabled and that diagnostics are on. Maybe we should have a specific function for that? But what's there is also fine.